### DISCUSSION TITLE
UpdateListing with DE texts, get EN back (also get listings seems is in english at the end )

### DISCUSSION THREAD

**Discussion by scuculeac-plenty** [etsy/open-api#868] [Q&A] [answered]
We are using updateListing call where we send title, description and so on in DE language but the response and the get listing are in english. What is the logic behine this? The exact same call on v2 worked fine but in v3 seems not correct.

We need urgent help since it affects a lot of clients at the moment.

Please check the images attached
<img width="1241" alt="Screenshot 2023-02-13 at 11 46 55" src="https://user-images.githubusercontent.com/82037958/218426240-3ebb73d6-c5d1-4d0e-9a23-36bf2f0079de.png">

<img width="1482" alt="Screenshot 2023-02-13 at 11 31 52" src="https://user-images.githubusercontent.com/82037958/218423445-231aff7e-6a7a-42a7-8a40-6d83bea00458.png">

<img width="1463" alt="Screenshot 2023-02-13 at 11 32 05" src="https://user-images.githubusercontent.com/82037958/218423452-1f465064-93a7-4209-959d-8b058816ece9.png">

**Comment by xaviered** [comment]
Hello,

Could you provide any header or params you are sending in your V2/V3?
Please strip out any confidential information like a token, API key, session, or cookie.

I will try to reproduce your issue to debug what's going on.

Thank you.

**Reply by scuculeac-plenty** [reply]
Thank you @xaviered  for you time and response. I am currently testing some scenarios and i will get back to you with more details!

Best Regards

**Reply by xaviered** [reply]
In the meantime, while we figure what's going on, you can also pass in `language=de` as a query parameter to force-get the `de` translation.

**Comment by scuculeac-plenty** [comment, marked as answer]
language=de as a query parameter to force-get the de translation -> this works if anyone needed resolved

**Comment by nateswart** [comment]
Following up on this thread:

We have [released a fix](https://github.com/etsy/open-api/releases/tag/3.0.0-general-release-2023-03-16) for this translation issue. All endpoints that return a `Listing` (not just `getListing`) will now default the language to  the shop’s default language.

Also, specifically on `getListing` you can override this behavior by passing in a `language` parameter. If the `Listing` does not support the language that was requested we return a 400 error with a message indicating the Listing translation does not exist.

**Reply by devtronic** [reply]
Can you also add the language parameter to the /listings/batch endpoint? Otherwise we have to request each product individually.
